cmd/agent/app/run.go

```diff
@@ -235,7 +235,12 @@ func StartAgent() error {
 	if config.Datadog.GetBool("telemetry.enabled") {
 		http.Handle("/telemetry", telemetry.Handler())
 	}
-	go http.ListenAndServe("127.0.0.1:"+port, http.DefaultServeMux) //nolint:errcheck
+	go func() {
+		err := http.ListenAndServe("127.0.0.1:"+port, http.DefaultServeMux)
+		if err != nil && err != http.ErrServerClosed {
+			log.Errorf("Error creating expvar server: %v", err)
```

Comment:
not absolutely necessary, but could be nice to log the bind address (host:port) explicitly in the error message (for all the messages you've added actually)

Reply:
Good point.